src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/directionissued/DirectionIssuedAboutToSubmitHandler.java

```diff
@@ -60,35 +64,128 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
         CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
         SscsCaseData caseData = caseDetails.getCaseData();
 
+        return validateDirectionType(caseData)
+                .or(()        -> validateDirectionDueDate(caseData))
+                .orElseGet(() -> validateForPdfAndCreateCallbackResponse(callback, caseDetails, caseData));
+    }
+
+    private Optional<PreSubmitCallbackResponse<SscsCaseData>> validateDirectionType(SscsCaseData caseData) {
         if (caseData.getDirectionTypeDl() == null || caseData.getDirectionTypeDl().getValue() == null) {
             PreSubmitCallbackResponse<SscsCaseData> errorResponse = new PreSubmitCallbackResponse<>(caseData);
             errorResponse.addError("Direction Type cannot be empty");
-            return errorResponse;
+            return Optional.of(errorResponse);
         }
+        return Optional.empty();
+    }
+
+    private Optional<PreSubmitCallbackResponse<SscsCaseData>> validateDirectionDueDate(SscsCaseData caseData) {
         if (DirectionType.PROVIDE_INFORMATION.toString().equals(caseData.getDirectionTypeDl().getValue().getCode())
                 && isBlank(caseData.getDirectionDueDate())) {
             PreSubmitCallbackResponse<SscsCaseData> errorResponse = new PreSubmitCallbackResponse<>(caseData);
             errorResponse.addError("Please populate the direction due date");
-            return errorResponse;
+            return Optional.of(errorResponse);
+        }
+        return Optional.empty();
+    }
+
+    private SscsCaseData updateCaseAfterExtensionRefused(SscsCaseData caseData, String interlocReviewState, State state) {
+        caseData.setHmctsDwpState("sentToDwp");
+        caseData.setDateSentToDwp(LocalDate.now().toString());
+        caseData.setInterlocReviewState(interlocReviewState);
+        caseData.setState(state);
+
+        return caseData;
+    }
+
+    @NotNull
+    private SscsCaseData updateCaseForDirectionType(CaseDetails<SscsCaseData> caseDetails, SscsCaseData caseData) {
+
+        if (DirectionType.PROVIDE_INFORMATION.toString().equals(caseData.getDirectionTypeDl().getValue().getCode())) {
+
+            caseData.setInterlocReviewState(AWAITING_INFORMATION.getId());
+
+        } else if (getPreValidStates().contains(caseDetails.getState())
+                && DirectionType.APPEAL_TO_PROCEED.toString().equals(caseData.getDirectionTypeDl().getValue().getCode())) {
+            caseData.setDateSentToDwp(LocalDate.now().toString());
+            caseData.setInterlocReviewState(AWAITING_ADMIN_ACTION.getId());
+
+        } else if (DirectionType.REFUSE_EXTENSION.toString().equals(caseData.getDirectionTypeDl().getValue().getCode())
+                && ExtensionNextEvent.SEND_TO_LISTING.toString().equals(caseData.getExtensionNextEventDl().getValue().getCode())) {
```

Comment:
worth doing a null check for getExtensionNextEventDl() and the following getValue